### PR TITLE
#473 Provide route hook timeout property

### PR DESCRIPTION
**Pull request by Kusig**
Provide the optional timeout property for route hook configuration to override the default of 30sec where necessary.

### CODE REVIEW

File: gateleen-hook/src/main/java/org/swisspush/gateleen/hook/HttpHook.java [hunk 1]
```diff
@@ -302,6 +304,24 @@ public void setMaxWaitQueueSize(Integer maxWaitQueueSize) {
         this.maxWaitQueueSize = maxWaitQueueSize;
     }
 
+    /**
+     * @return Timeout for request. This may returning null in case there's no value
+     *         specified. Callers may catch that by fall back to a default.
+     */
+    public Integer getTimeout() {
```
Comment (dominik-cnx) [inline gateleen-hook/src/main/java/org/swisspush/gateleen/hook/HttpHook.java:311]: Propose to annotate with @Nullable

File: gateleen-hook/src/main/java/org/swisspush/gateleen/hook/HttpHook.java [hunk 2]
```diff
@@ -302,6 +304,24 @@ public void setMaxWaitQueueSize(Integer maxWaitQueueSize) {
         this.maxWaitQueueSize = maxWaitQueueSize;
     }
 
+    /**
```
Comment (dominik-cnx) [inline gateleen-hook/src/main/java/org/swisspush/gateleen/hook/HttpHook.java:307]: Should we further describe the timeout here, or link to a detailed description?

File: gateleen-hook/src/main/java/org/swisspush/gateleen/hook/HttpHook.java (hunk 1 above)
Comment (Kusig) [reply gateleen-hook/src/main/java/org/swisspush/gateleen/hook/HttpHook.java:311]: Will do so.

File: gateleen-hook/src/main/java/org/swisspush/gateleen/hook/HttpHook.java (hunk 2 above)
Comment (Kusig) [reply gateleen-hook/src/main/java/org/swisspush/gateleen/hook/HttpHook.java:307]: Will extend it.

**Review by dominik-cnx — APPROVED** [review]

File: gateleen-hook/src/main/java/org/swisspush/gateleen/hook/HttpHook.java [hunk 3]
```diff
@@ -302,6 +305,27 @@ public void setMaxWaitQueueSize(Integer maxWaitQueueSize) {
...
+    public Integer getTimeout() {
+        return this.timeout;
+    }
+
+    /**
+     * See {@link #getTimeout()}.
+     */
+    public void setTimeout(Integer timeout) {
+        if (timeout != null && timeout < -1){
+            throw new IllegalArgumentException("Values below -1 not valid.");
```
Comment (mcweba) [inline gateleen-hook/src/main/java/org/swisspush/gateleen/hook/HttpHook.java:324]: Is a timeout value of **-1** allowed? Should it not always be a positive number?

Comment (Kusig) [reply gateleen-hook/src/main/java/org/swisspush/gateleen/hook/HttpHook.java]: Right, will improve this.

**Review by mcweba — APPROVED** [review]

**Review by hiddenalpha — APPROVED** [review]
(*just curious*) I saw lots of code make this timeout configurable and hand it over in misc places. But I could not find any place where it actually is used.

I assume this is by intent?

**Review by ljucam — APPROVED** [review]

**Comment by Kusig** [comment]
> (_just curious_) I saw lots of code make this timeout configurable and hand it over in misc places. But I could not find any place where it actually is used.
> 
> I assume this is by intent?

> (_just curious_) I saw lots of code make this timeout configurable and hand it over in misc places. But I could not find any place where it actually is used.
> 
> I assume this is by intent?

This timeout is not new, it is already used in the Rule.java when setting up the HttpClient. It was just fix on 30 seconds before and not configurable. With this PR it becomes optionally configurable via the Hook configuration.

**Pull request merged**